src/inmanta/ast/type.py

```diff
@@ -387,6 +390,9 @@ def validate(self, value: Optional[object]) -> bool:
     def type_string_internal(self) -> str:
         return "List"
 
+    def type_string(self) -> str:
+        return "list"
```

Comment:
I'm somewhat concerned about this change. I have to look into it.

Reply:
That would be nice, I didn't know how this was intended

Reply:
So the issue is that this method is meant to represent a mapping between a DSL type and its string representation. But not all children of `Type` are DSL types, e.g. `Union` is an internal type used to represent primitives, `List` is an internal type, with `TypedList` and `LiteralList` its concrete DSL types. If I had to rewrite the whole typing hierarchy I'd probably do it differently, but that's beside the point now. I think we should keep this method as is for now. For any user-definable attribute type, the method should return a not-`None` value, so you could go with an assertion.

Reply:
so you propose to not override this here, as it is a ABC, and only override it in the concrete leaf types?